libs/libcommon/src/libcommon/queue.py

```diff
@@ -600,6 +609,47 @@ def get_dataset_pending_jobs_for_type(self, dataset: str, job_type: str) -> List
             )
         ]
 
+    def heartbeat(self, job_id: str) -> None:
+        job = Job.objects.with_id(job_id)  # type: ignore
+        if job is not None and isinstance(job, Job):
+            job.update(last_heartbeat=get_datetime())
+
+    def get_zombies(self, max_seconds_without_heartbeat: int) -> List[JobInfo]:
+        started_jobs = Job.objects(status=Status.STARTED)
+        if max_seconds_without_heartbeat > 0:
+            zombies = [
+                job
+                for job in started_jobs
+                if (
+                    job.last_heartbeat is not None
+                    and get_datetime()
+                    >= pytz.UTC.localize(job.last_heartbeat) + timedelta(seconds=max_seconds_without_heartbeat)
+                )
+                or (
+                    job.last_heartbeat is None
+                    and job.started_at is not None
+                    and get_datetime()
+                    >= pytz.UTC.localize(job.started_at) + timedelta(seconds=max_seconds_without_heartbeat)
+                )
+            ]
+            return [zombie.info() for zombie in zombies]
+        else:
+            return []
+
+    def kill_zombies(self, zombies: List[JobInfo]) -> int:
+        if zombies:
```

Comment:
same nit:

```
if not zombies:
  return 0
zombie_job_ids = [zombie["job_id"] for zombie in zombies]
...

Reply:
btw it's what [Sourcery](https://marketplace.visualstudio.com/items?itemName=sourcery.sourcery) proposes:

> Sourcery - Swap if/else branches, Remove unnecessary else after guard condition (refactoring:swap-if-else-branches;remove-unnecessary-else)